common/static/common/js/components/views/tabbed_view.js

```diff
@@ -1,138 +1,239 @@
+            },
+            
+            /**
+            * View for a tabbed interface. Expects a list of tabs
+            * in its options object, each of which should contain the
+            * following properties:
+            *   view (Backbone.View): the view to render for this tab.
+            *   title (string): The title to display for this tab.
+            *   url (string): The URL fragment which will
+            *     navigate to this tab when a router is
+            *     provided.
+            * If a router is passed in (via options.router),
+            * use that router to keep track of history between
+            * tabs.  Backbone.history.start() must be called
+            * by the router's instantiator after this view is
+            * initialized.
+            */
+            initialize: function (options) {
+                this.router = options.router || null;
+                this.tabs = options.tabs;
+                this.template = _.template(tabbedViewTemplate)({
+                    viewLabel: this.viewLabel
+                });
+                // Convert each view into a TabPanelView
+                _.each(this.tabs, function(tabInfo, index) {
+                    tabInfo.view = new TabPanelView({
+                        url: tabInfo.url,
+                        view: tabInfo.view,
+                        index: index
+                    });
+                }, this);
+                this.urlMap = _.reduce(this.tabs, function (map, value) {
+                    map[value.url] = value;
+                    return map;
+                }, {});
+            },
+            render: function () {
+                var self = this;
+                this.$el.html(this.template);
+                _.each(this.tabs, function(tabInfo, index) {
+                    var tabEl = $(_.template(tabTemplate)({
+                        index: index,
+                        title: tabInfo.title,
+                        url: tabInfo.url,
+                        tabPanelId: getTabPanelId(tabInfo.url)
+                    })),
+                    tabContainerEl = this.$('.tabs');
+                    self.$('.page-content-nav').append(tabEl);
+                    
+                    // Render and append the current tab panel
+                    tabContainerEl.append(tabInfo.view.render().$el);
+                }, this);
+                // Re-display the default (first) tab if the
+                // current route does not belong to one of the
+                // tabs.  Otherwise continue displaying the tab
+                // corresponding to the current URL.
+                if (!(Backbone.history.getHash() in this.urlMap)) {
+                    this.setActiveTab(0);
+                }
+                
+                return this;
+            },
+            
+            setActiveTab: function(index) {
+                var $tabMeta = this.getTabMeta(index),
+                    $tab = $tabMeta.tab,
```

Comment:
Only use the ``$`` prefix for JQuery elements. This should just be ``tab`` and the one below should be ``$tab``.